src/core_modules/capture-core/components/DataEntry/DataEntry.component.tsx

```diff
@@ -181,11 +200,13 @@ class DataEntryPlain extends React.Component<Props> {
             )
         );
 
-    handleUpdateFieldAsync = (...args) => {
-        this.props.onUpdateFormFieldAsync(...args, this.props.id, this.props.itemId);
+    handleUpdateFieldAsync = (...args: any[]) => {
+        if (this.props.onUpdateFormFieldAsync && this.props.itemId) {
+            this.props.onUpdateFormFieldAsync(...args as [string, string, string, string, (...callbackArgs: any[]) => void], this.props.id, this.props.itemId);
```

Comment:
Change this line to `this.props.onUpdateFormFieldAsync(...args, this.props.id, this.props.itemId);` and fix the error by using `...args: [string, string, string, string, (...callbackArgs: any[]) => void]` on line 203